src/wstool/multiproject_cli.py

```diff
@@ -621,6 +623,59 @@ def cmd_diff(self, target_path, argv, config=None):
 
         return False
 
+    def cmd_foreach(self, target_path, argv, config=None):
+        """Run shell commands in each repository."""
+        parser = argparse.ArgumentParser(
+            usage='usage: %s foreach [command] [OPTIONS]' % self.progname,
+            description=__MULTIPRO_CMD_DICT__['foreach'],
+            epilog='See: http://www.ros.org/wiki/rosinstall for details')
+        parser.add_argument('command', type=str, help='command to run')
+        parser.add_argument('--shell', default=False,
```

Comment:
Maybe query for localname will be necessary, but I will add only that for scm at this point.